test/common/http/http2/codec_impl_test.cc

```diff
@@ -489,21 +503,26 @@ TEST_P(Http2CodecImplTest, TrailerStatus) {
   HttpTestUtility::addDefaultHeaders(request_headers);
   EXPECT_CALL(request_decoder_, decodeHeaders_(_, true));
   EXPECT_TRUE(request_encoder_->encodeHeaders(request_headers, true).ok());
+  driveToCompletion();
+
   EXPECT_TRUE(Http2CodecImplTestFixture::slowContainsStreamId(1, *client_));
   EXPECT_FALSE(Http2CodecImplTestFixture::slowContainsStreamId(100, *client_));
 
   TestResponseHeaderMapImpl continue_headers{{":status", "100"}};
   EXPECT_CALL(response_decoder_, decode1xxHeaders_(_));
   response_encoder_->encode1xxHeaders(continue_headers);
+  driveToCompletion();
 
   TestResponseHeaderMapImpl response_headers{{":status", "200"}};
   EXPECT_CALL(response_decoder_, decodeHeaders_(_, false));
   response_encoder_->encodeHeaders(response_headers, false);
+  driveToCompletion();
```

Comment:
should we be checking both statuses are OK after every call to driveToCompletion() where you are not expecting them to *not* be OK?

Reply:
I'm open to this idea - we could have driveToCompletion() return a bool on whether both statuses are OK, and then update the call sites accordingly to EXPECT_TRUE(driveToCompletion()). At the same time, this would be adding new expectations (IIUC, the original tests .IgnoreError() by default). I can add them IFF they're helpful signal. Thanks!